### PR TITLE
improve training stability (AdamW, BCE, RMSE)

### PR DESCRIPTION
**Pull request by IamMuhammadZeeshan**
**Summary:**

This PR introduces significant improvements to the neural regressor training routine for pdet prediction.
The goal is to make training smoother, more stable, and numerically consistent while providing interpretable metrics (such as RMSE) for better model evaluation.

**Key Changes:**

	•	Optimizer: Switched from plain Adam to AdamW for improved generalization and weight regularization.
	•	Learning Rate Scheduler: Added cosine decay with warmup to gradually ramp up learning and prevent sharp loss oscillations.
	•	Gradient Clipping: Applied clip_by_global_norm to prevent gradient explosions and reduce loss spikes.
	•	Loss Function Options:
	•	Added support for Binary Cross-Entropy with logits (bce_logits) — suitable for training on [0,1] probability targets such as pdet.
	•	Retained existing MSE loss for regression-style experiments.
	•	Validation Split: Now uses a fixed random seed to ensure reproducible train/validation splits.
	•	RMSE Metric: Added Root Mean Square Error logging on the probability scale (after sigmoid), giving interpretable feedback even when BCE loss remains high.
	•	Logging: Expanded progress logging to include key training hyperparameters and evaluation results.
	•	Plots: Retained the familiar _loss.png and _total_loss.png diagnostics for consistency.
	
**Motivation:**

The previous implementation used plain Adam with a constant learning rate and unbounded gradients, which often resulted in spiky validation losses and unstable convergence.
This new setup yields:
	•	smoother learning curves,
	•	lower validation variance,
	•	improved interpretability of the final model’s performance on pdet predictions.

### CODE REVIEW

File: src/gwkokab/utils/train.py [hunk 1]
```diff
@@ -207,52 +141,59 @@ def load_model(filename: str) -> Tuple[List[str], eqx.nn.MLP]:
...
             layer_i = f[f"layer_{i}"]
             weight_i = jax.device_put(np.asarray(layer_i[f"weight_{i}"][:]))
             bias_i = jax.device_put(np.asarray(layer_i[f"bias_{i}"][:]))
-            new_model = eqx.tree_at(lambda m: m.layers[i].weight, new_model, weight_i)
-            new_model = eqx.tree_at(lambda m: m.layers[i].bias, new_model, bias_i)
+            new_model = eqx.tree_at(
+                lambda m, ii=i: m.layers[ii].weight, new_model, weight_i
+            )
+            new_model = eqx.tree_at(
+                lambda m, ii=i: m.layers[ii].bias, new_model, bias_i
```
Comment (Qazalbash) [inline src/gwkokab/utils/train.py]: The `ii` argument is unnecessary.

File: src/gwkokab/utils/train.py [hunk 2]
```diff
@@ -268,204 +209,217 @@ def train_regressor(
     validation_split: float = 0.2,
     learning_rate: float = 1e-3,
     train_in_log: bool = False,
+    # New, optional stabilizers (keep old calls working):
+    loss_type: str = "mse",  # "mse" or "bce_logits"
+    grad_clip_norm: float = 1.0,
+    weight_decay: float = 1e-4,
+    use_cosine_decay: bool = True,
+    min_lr: float = 1e-6,
```
Comment (Qazalbash) [inline src/gwkokab/utils/train.py]: It should be changed to the appropriate full form.

**Review by Qazalbash — CHANGES_REQUESTED** [review]
Good changes. I will still prefer to use mean-squared error, but we can experiment with other losses too.
